test/E2ETest/Microsoft.Test.E2E.AspNet.OData/BoundOperation/BoundOperationContextUriTest.cs

```diff
@@ -0,0 +1,224 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.AspNet.OData;
+using Microsoft.AspNet.OData.Builder;
+using Microsoft.AspNet.OData.Extensions;
+using Microsoft.OData.Edm;
+using Microsoft.Test.E2E.AspNet.OData.Common.Controllers;
+using Microsoft.Test.E2E.AspNet.OData.Common.Execution;
+using Xunit;
+
+namespace Microsoft.Test.E2E.AspNet.OData.BoundOperation
+{
+    public class BoundOperationContextUriTest : WebHostTestBase
+    {
+        public BoundOperationContextUriTest(WebHostTestFixture fixture)
+            : base(fixture)
+        {
+        }
+
+        protected override void UpdateConfiguration(WebRouteConfiguration configuration)
+        {
+            var controllers = new[] { typeof(ProjectsController) };
+            configuration.AddControllers(controllers);
+            configuration.Routes.Clear();
+
+            // for Operation context Uri testing
+            configuration.MapODataServiceRoute("odata", "odata", GetEdmModel());
+        }
+
+        [Fact]
+        public async Task BoundFunctionWithNonContainmentReturnTypeWorksWithCorrectContextUri()
+        {
+            // Arrange
+            var requestUri = string.Format("{0}/odata/Projects/Default.GetProjectTask", BaseAddress);
```

Comment:
>            var requestUri = string.Format("{0}/odata/Projects/Default.GetProjectTask" [](start = 0, length = 86)

how about adding a segment after the bound function call, i.e.: 
-Projects/Default.GetProjectTask/1
-Projects/Default.GetDefaultProjectTask/Default.ProjectTask